ibis/backends/sql/compilers/mssql.py

```diff
@@ -526,5 +524,23 @@ def visit_StartsWith(self, op, *, arg, start):
     def visit_EndsWith(self, op, *, arg, end):
         return arg.like(self.f.concat("%", end))
 
+    def visit_LPad(self, op, *, arg, length, pad):
+        return sge.Case(
+            ifs=[self.if_(length <= self.f.length(arg), arg)],
+            default=self.f.left(
+                self.f.concat(self.f.replicate(pad, length - self.f.length(arg)), arg),
+                length,
+            ),
+        )
+
+    def visit_RPad(self, op, *, arg, length, pad):
+        return sge.Case(
+            ifs=[self.if_(length <= self.f.length(arg), arg)],
+            default=self.f.left(
+                self.f.concat(arg, self.f.replicate(pad, length - self.f.length(arg))),
+                length,
+            ),
+        )
```

Comment:
These can both be simplified to `return self.if_(...)` since there's only a single truth case and a default, but that can be done in a follow-up.

Reply:
Thank you for guiding this simplification and approving it in #10085!